transformers/storage/median/repository.go

```diff
@@ -0,0 +1,286 @@
+		switch metadata.PackedNames[order] {
+		case Val:
+			insertErr = repository.insertMedianVal(diffID, headerID, contractAddress, value)
+		case Age:
+			insertErr = repository.insertMedianAge(diffID, headerID, contractAddress, value)
+		default:
+			panic(fmt.Sprintf("unrecognized median contract storage name in packed values: %s", metadata.Name))
+		}
+		if insertErr != nil {
+			return insertErr
+		}
+	}
+	return nil
+}
+
+func (repository *MedianStorageRepository) insertOrcl(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, orcl string) error {
+	orclAddress, orclErr := getOrclAddress(metadata.Keys)
+	if orclErr != nil {
+		return orclErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+
+	orclAddressID, orclAddressErr := shared.GetOrCreateAddress(orclAddress, repository.db)
+	if orclAddressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("median orcl address", orclAddressErr)
+		}
+		return addressErr
+	}
+
+	_, insertErr := tx.Exec(insertMedianOrclQuery, diffID, headerID, addressID, orclAddressID, orcl)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bud record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func getOrclAddress(keys map[types.Key]string) (string, error) {
+	user, ok := keys[constants.Address]
+	if !ok {
+		return "", types.ErrMetadataMalformed{MissingData: constants.A}
+	}
+	return user, nil
+}
+
+func (repository *MedianStorageRepository) insertBud(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, bud string) error {
+	budAddress, budErr := getBudAddress(metadata.Keys)
+	if budErr != nil {
+		return budErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+
+	budAddressID, budAddressErr := shared.GetOrCreateAddress(budAddress, repository.db)
+	if budAddressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("median bud address", budAddressErr)
+		}
+		return addressErr
+	}
+
+	_, insertErr := tx.Exec(insertMedianBudQuery, diffID, headerID, addressID, budAddressID, bud)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bud record with address", insertErr)
+		}
+		return insertErr
```

Comment:
similarly nbd but could be cool to use `fmt.Errorf` throughout - so like here we could do:
```
return fmt.Errorf("error inserting median bud for address %s: %w", contractAddress, insertErr)
```
mostly just helps out with debugging so you don't get stack traces with errors like `error executing transformer: sql no rows` without intermediate context